src/test/java/dev/openfeature/sdk/testutils/Flag.java

```diff
@@ -0,0 +1,18 @@
+package dev.openfeature.sdk.testutils;
+
+import io.cucumber.core.internal.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import java.util.Map;
+
+@ToString
+@JsonIgnoreProperties(ignoreUnknown = true)
+@NoArgsConstructor
+@Getter
+public class Flag {
+    private Flags.State state;
+    private Map<String, Object> variants;
+    private String defaultVariant;
+}
```

Comment:
Trying to understand what is the purpose and value here with enhancing with additional testing capabilities. The in-memory provider is for testing purposes, how is adding a context evaluator helping with testing actual flow ? it will only test the testing provider ?

Reply:
The current implementation resides as a `test` source. This mismatch with the requirement of the specification. 

When we agreed on the in-memory provider through OFEP and added it to the specification [^1], the agreement was to make the in-memory an extra provider built into the SDK itself. And we agreed to support evaluation contexts through lambda/callback.

So the packaging of the implementation should be corrected first. I am proposing to move it to a package named `memprovider` or any better-named package.

Regarding the purpose, the main benefit of having evaluation context support is testing SDK. It allows us to write end-to-end tests (or to migrate existing ones based on flagd) for context evaluations and verify SDK correctness. Besides, end users can use the provider to prototype OpenFeature features. 




[^1]: https://github.com/open-feature/spec/blob/main/specification/appendix-A.md#in-memory-provider